src/main/java/com/demo/api/controller/StaffController.java

```diff
@@ -2,30 +2,56 @@
 
 import java.util.*;
 
+// import org.apache.ibatis.annotations.Param;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
-import com.demo.api.model.*;
+import org.springframework.web.bind.annotation.RequestBody;
+
+
+import com.demo.api.entities.StaffEntity;
 import com.demo.api.service.StaffService;
 
+
+/**
+ * StaffController
+ * 
+ * @author QuynhNN
+ */
 @RestController
 @RequestMapping("/staff")
 public class StaffController {
 
     @Autowired
     private StaffService staffService;
 
-    // test method
-    @GetMapping(value = "/get")
-    public ResponseEntity<String> get() {
-        return ResponseEntity.ok("test");
+    /**
+     * Get details staff by id.
+     * @param id the id of the staff in database
+     * @return staff details
+     */
+    @GetMapping(value = "/getStaffDetailsById/{id}")
+    public StaffEntity getStaffDetailsById(@PathVariable int id) {
+        return staffService.getStaffDetailsById(id);
+    }
+    
+    /**
+     * Get all staffs in database
+     * @return all staff
+     */
+    @GetMapping(value = "/getAll")
+    public List<StaffEntity> getAllStaffs() {
+        return staffService.getAllStaff();
     }
 
-    // get all staffs
-    @GetMapping(value = "/staffs")
-    public List<Staff> getAllStaffs() {
-        return staffService.findAll();
+    /**
+     * update a staff by id
+     */
+    @GetMapping(value="/update/{id}")
+    public void updateStaffById(@PathVariable(value = "id") int id, @RequestBody StaffEntity entity) {
```

Comment:
tạo class request, không truyền trực tiếp entity

Reply:
em đã cập nhật lại: 
StaffController.java
@PutMapping(value = "/update/{id}")
    public void updateStaffById(@PathVariable(value = "id") int id, @RequestBody StaffRegisterRequest staffRegisterRequest) {
        staffService.updateStaffById(staffRegisterRequest, id);
    }

StaffService.java
public void updateStaffById(StaffRegisterRequest staffRegisterRequest, int id) {
        StaffEntity staffEntity = new StaffEntity();
        staffEntity.setName(staffRegisterRequest.getName());
        staffEntity.setAddress(staffRegisterRequest.getAddress());
        staffEntity.setPhoneNumber(staffRegisterRequest.getPhoneNumber());
        staffEntity.setDateOfBirth(staffRegisterRequest.getDateOfBirth());
        this.staffRepository.updateStaffById(staffEntity, id);
    }